lib/CMakeLists.txt

```diff
@@ -310,6 +310,8 @@ target_include_directories(syslog-ng
 target_include_directories(syslog-ng
   SYSTEM PUBLIC
     ${CORE_INCLUDE_DIRS}
+  PRIVATE
```

Comment:
The "missing" include directories are not a bug but a feature of cmake: https://github.com/syslog-ng/syslog-ng/pull/2007

Reply:
The mentioned compile error shows that the initial assumption in #2007 was wrong. Setting `target_link_libraries` does not set `include_directories` for `Openssl:SSL`.

Reply:
It's not an assumption, it's a design decision in modern CMake, so you can "advertise" your own build specifications; transitive dependencies, include paths, and compiler flags without creating boilerplate and error-prone constructs on the user side.

Basically you define "packages" (targets) instead of random variables, and on the user side, you only have to use `target_link_libraries()`, which has very little to do with the low-level "linking" concept. (Please read #2007, [1](https://cmake.org/cmake/help/latest/manual/cmake-buildsystem.7.html#build-specification-and-usage-requirements), [2](https://pabloariasal.github.io/2018/02/19/its-time-to-do-cmake-right/)).

`OpenSSL::SSL` is an imported target that uses the same modern concept, and therefore `include_directories()` is clearly set:
https://github.com/Kitware/CMake/blob/master/Modules/FindOpenSSL.cmake#L502-L503

Reply:
Ok, I understand your motivation. I try to find another resolution. Adding WIP.